dashboard/app/controllers/registrations_controller.rb

```diff
@@ -26,6 +26,22 @@ def new
   end
 
   #
+  # POST /users/begin_sign_up
+  #
+  # Submit step 1 of the signup process for creating an email/password account.
+  #
+  def begin_sign_up
+    @user = User.new(begin_sign_up_params)
+    @user.validate_for_finish_sign_up
+
+    if @user.errors.blank?
```

Comment:
This 'partial validation' pattern for our partial registration is interesting and seems a little messy... but I don't know what I'd do instead.  Run `valid?` and then filter the resulting errors to fields we care about? That wouldn't be particularly efficient.  `PartialRegistration` isn't a model but if it was it might make sense for it to own validation - but then, we'd almost certainly be duplicating logic between it and `User`.

No action here, just generally unsatisfied with all options. :confounded:

Reply:
yeah, i was having _a lot_ of trouble trying to figure out the best way to do this. i even have to update our `password_required?` validation (in a future PR) in order for this to work 😢 

the `validate_for_finish_sign_up` method does something similar to your suggestion -- run all the validations, then remove ones we don't care about 🤷‍♀️ i'm still trying to think of ways to improve it. i was previously calling validations by name, but it was getting way too crazy